### PR TITLE
OOP school library: preserve data

### PR DESCRIPTION
**Pull request by herokudev**
In this project, you will preserve the data from your app and load preserved data on startup.
-     Store data in files.
-     Run a program using the command line.

### CODE REVIEW

File: library/rental.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  end
+
+  def to_json(*args)
+    {
+      JSON.create_id => self.class.name,
+      'date' => @date,
+      'book_title' => @book.title,
+      'person_id' => @person.id
+    }.to_json(*args)
+  end
```
Comment (Dipeshtwis) [inline library/rental.rb]: - [x] After creating a rental, when I am trying to fetch the rental, the application crashes, and also it does not preserve data. Please fix that.

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @herokudev and @leonardoeng13, 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.
:+1: The person and book data are preserved well.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


 ## Required Changes ♻️

**_Check the comments under the review_**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- N.A.

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**_Dipesh Kumar_**


------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: library/rental.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+  def to_json(*args)
+    {
+      JSON.create_id => self.class.name,
+      'date' => @date,
+      'book_title' => @book,
+      'person_id' => @person
+    }.to_json(*args)
+  end
+end
```
Comment (henatan99) [inline library/rental.rb]: - [x] The previous reviewer pointed out here that the data is not preserved. Unfortunately, this issue is not addressed and all of the data (people, books, and rentals) are not preserved after being created by the user. In addition, while the app runs it should parse data from the JSON files but currently, it is using the data in the current session. Kindly fix these issues. :pray:

**Review by henatan99 — CHANGES_REQUESTED** [review]
Hi @herokudev , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: library/rental.rb (hunk 2 above)
Comment (leonardoeng13) [reply library/rental.rb]: Hi. I must have to disagree with you about the data preserved. All data are preserved. We uploaded the files with some data to make more "easy" for you to test, and of course, you're able to create new data. And I would like to add more information on this project request: **On exit** from your app save all the data (books, people, and rentals) into JSON files. One per type of data you are storing and named accordingly (e.g. rentals.json)" It's mean to save the data in files after exiting the program. I would like to rest you to test again and change from "Change request" to approved.

File: README.md [hunk 3]
```diff
@@ -2,10 +2,17 @@
...
 
 This repository aims to practice OOP(Object Oriented Programming) for best practice. App School Library was made with Ruby Language.
 
+
+## Getting Started
+
+This repository is located on Github and you can get a copy with this link [here](https://github.com/leonardoeng13/OOP-school-library.git)
+
+In your terminal, go to the source code folder and execute "ruby ./library/main.rb".
+
```
Comment (ZeenatLawal) [inline README.md:15]: - [ ] Kindly update the execute command before merging.

**Review by ZeenatLawal — APPROVED** [review]
Hi @leonardoeng13 and @herokudev, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

## Congratulations! 🎉 
6vEo


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**